salt/tests/unit/modules/test_metalk8s_monitoring.py

```diff
@@ -0,0 +1,185 @@
+from datetime import datetime
+import os.path
+
+from salt.exceptions import CommandExecutionError
+
+from parameterized import param, parameterized
+from salttesting.helpers import ForceImportErrorOn
+from salttesting.mixins import LoaderModuleMockMixin
+from salttesting.mock import MagicMock, patch
+from salttesting.unit import TestCase
+import yaml
+
+import metalk8s_monitoring
+
+
+YAML_TESTS_FILE = os.path.join(
+    os.path.dirname(os.path.abspath(__file__)),
+    "files", "test_metalk8s_monitoring.yaml"
+)
+with open(YAML_TESTS_FILE) as fd:
+    YAML_TESTS_CASES = yaml.safe_load(fd)
+
+
+def _custom_name_func(testcase_func, _, param):
+    return '{}_{}'.format(
+        testcase_func.__name__,
+        param.kwargs['_id'].replace('-', '_')
+    )
```

Comment:
Note to reviewers: that's merely a test for making verbose pytest output more readable and useful.
Ideally, this custom naming + whatever technique we use for extracting test cases from YAML files should be factorized and standardized.